[16.0][l10n_br_account_payment_order][l10n_br_account_payment_brcobranca] unify cnab code objects

rvalyi opened the pull request:
port de https://github.com/OCA/l10n-brazil/pull/3337

mileo committed:
[NEW] Módulo de relatórios contabeis

Signed-off-by: Luis Felipe Mileo <mileo@kmee.com.br>

mileo committed:
[NEW] Some fixes and report styles

Signed-off-by: Luis Felipe Mileo <mileo@kmee.com.br>

Signed-off-by: Luis Felipe Mileo <mileo@kmee.com.br>

Signed-off-by: Luis Felipe Mileo <mileo@kmee.com.br>

DiegoParadeda committed:
[ADD] MIS Reports for simples nacional

mileo committed:
[NEW] DRE auto expand accounts

Signed-off-by: Luis Felipe Mileo <mileo@kmee.com.br>

mileo committed:
[NEW] OCA readme files

Signed-off-by: Luis Felipe Mileo <mileo@kmee.com.br>

mileo committed:
[FIX] DRE expression CSLL e IRPJ

Signed-off-by: Luis Felipe Mileo <mileo@kmee.com.br>

mileo committed:
[REF] BP with new account.account.types

Signed-off-by: Luis Felipe Mileo <mileo@kmee.com.br>

mileo committed:
[FIX] l10n_br_mis_report dependencies

Signed-off-by: Luis Felipe Mileo <mileo@kmee.com.br>

mileo committed:
[FIX

mileo committed:
[FIX] Mis Report

Signed-off-by: Luis Felipe Mileo <mileo@kmee.com.br>

mileo committed:
[FIX] DRE e BP images

Signed-off-by: Luis Felipe Mileo <mileo@kmee.com.br>

mileo committed:
[FIX] wrong bp expression

Signed-off-by: Luis Felipe Mileo <mileo@kmee.com.br>

oca-travis committed:
[UPD] Update l10n_br_mis_report.pot

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
l10n_br_mis_report 12.0.1.1.0

rvalyi committed:
l10n_br_mis_report: fixed images

OCA-git-bot committed:
[UPD] README.rst

marcelsavegnago committed:
Added translation using Weblate (Portuguese (Brazil))

mileo committed:
[FIX] DRE deduções da receita

OCA-git-bot committed:
l10n_br_mis_report 12.0.1.2.0

rvalyi committed:
DRE: bale->balp everywhere

OCA-git-bot committed:
l10n_br_mis_report 12.0.1.3.0

rvalyi committed:
[IMP] l10n_br_mis_report: black, isort, prettier

rvalyi committed:
[MIG] l10n_br_mis_report: Migration to 14.0

douglascstd committed:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 0.0% (0 of 52 strings)

Translation: l10n-brazil-14.0/l10n-brazil-14.0-l10n_br_mis_report
Translate-URL: https://translation.odoo-community.org/projects/l10n-brazil-14-0/l10n-brazil-14-0-l10n_br_mis_report/pt_BR/

mileo committed:
[REF] deprecade data xml node l10n_br_mis_report

OCA-git-bot committed:
l10n_br_mis_report 14.0.1.0.1

rvalyi committed:
[REF] cleaner db wise mixin mappings

mbcosta committed:
[FIX] l10n_br_fiscal: proper partner_id

When mapping the Line Fiscal Operation and Taxes the Partner of the object can be or not the Partner to Invoice/Fiscal.

[FIX] l10n_br_fiscal: Document get Fiscal Partner

mbcosta committed:
[FIX] l10n_br_sale: proper partner_id

Invoice should be create with partner_invoice_id not with partner_id.

[FIX] l10n_br_sale: Document get Fiscal Partner

[FIX] l10n_br_sale: Related Fiscal Partner

[FIX] l10n_br_sale: partner_invoice_id fix

When mapping the Line Fiscal Operation and Taxes the Partner of the object can be or not the Partner to Invoice, in case of Sale should it use the field partner_invoice_id.

mbcosta committed:
[FIX] l10n_br_stock_account: partner_shipping_id

when creating an Invoice use the Partner mapping by the methods of module instead of Fiscal Dict and inform partner_shipping_id with Partner of Picking instead in the case of it's different of Partner to Invoice.

CristianoMafraJunior committed:
[IMP] l10n_br_nfe: DANFE margins config

rvalyi committed:
[REF] extract l10n_br_fiscal_edi <- l10n_br_fiscal

[REF] extract l10n_br_fiscal_edi from l10n_br_fiscal 2/2

[TMP] leave invalidate.number in l10n_br_fiscal

[REF] l10n_br_fiscal: mv doc wrkflw ->fisc edi

[REF] l10n_br_fiscal_edi: mv doc wrkflw ->fisc edi

[WIP] mv edoc import to fiscal_edi

[REF] add fiscal document hook

[FIX] l10n_br_fiscal_edi:  wrong status_description

[FIX] l10n_br_fiscal: edi migration

mileo committed:
[MIG] l10n_br_mis_report: Migration to 15.0

mileo committed:
[IMP] l10n_br_mis_report: apply pre-commit auto fixes

mileo committed:
[MIG] l10n_br_mis_report: Migration to 16.0

rvalyi committed:
[REF] spec_driven_model: simplify register hook

rvalyi committed:
[REF] l10n_br_nfe: adapt to simpler spec_driven_model hook

rvalyi committed:
[MIG] l10n_br_account_nfe: Migration to 16.0

rvalyi committed:
[REF] l10n_br_account: adapt to l10n_br_fiscal_edi

rvalyi committed:
[REF] l10n_br_nfse_focus: adapt to l10n_br_fiscal_edi

mbcosta committed:
[FIX] l10n_br_purchase: proper partner_id

When mapping the Line Fiscal Operation and Taxes the Partner of the object can be or not the Partner to Invoice, in the Purchase case the code use address_get to create Invoice.

[FIX] l10n_br_purchase: Document get Fiscal Partner

mbcosta committed:
[IMP] l10n_br_purchase: demo data and tests

for the cases where the Partner of the object can be or not the Partner to Invoice.

mbcosta committed:
[IMP] l10n_br_sale: demo data and tests

for the case where the Partner of the object can be or not the Partner to Invoice.

mbcosta committed:
[FIX] l10n_br_stock_account: proper partner_id

When mapping the Line Fiscal Operation and Taxes the Partner of the object can be or not the Partner to Invoice, in the Picking case the code use address_get to create Invoice.

[FIX] l10n_br_stock_account: Get Fiscal Partner

mbcosta committed:
[IMP] l10n_br_stock_account: demo data and tests

for the case where the Partner of the object can be or not the Partner to Invoice.

rvalyi committed:
[REF] l10n_br_nfe: depend on l10n_br_fiscal_edi

rvalyi committed:
[REF] l10n_br_nfse: depend on l10n_br_fiscal_edi

rvalyi committed:
[REF] l10n_br_fiscal_closing: depend on l10n_br_fiscal_edi

corredato committed:
[NEW] CNPJ search in CRM Lead

mbcosta committed:
[FIX] l10n_br_stock_account: Get Default OP Fiscal

rvalyi committed:
[REF] l10n_br_fiscal: document_fiscal* rename

rvalyi committed:
[REF] l10n_br_account: fiscal_document* rename

rvalyi committed:
[REF] l10n_br_sale: fiscal_document rename

rvalyi committed:
[REF] l10n_br_account_nfe: adapt to simpler hook

mbcosta committed:
[REF] l10n_br_account_payment_order: Join CNAB Cod

mbcosta committed:
[REF] l10n_br_account_payment_order: View CNAB Cod

mbcosta committed:
[REF] l10n_br_account_payment_order: Data CNAB Cod

mbcosta committed:
[REF] l10n_br_account_payment_order: Demo CNAB Cod

mbcosta committed:
[REF] l10n_br_account_payment_order: Test CNAB Cod

mbcosta committed:
[REF] l10n_br_account_payment_brcobranca: Join CNAB Cod

antoniospneto committed:
[MIG] l10n_br_account_nfe: improve tests

DiegoParadeda committed:
[IMP] l10n_br_nfe: set nfe40_IEST

antoniospneto committed:
[REF] l10n_br_mis_report: pre-commit update

antoniospneto committed:
[MIG] l10n_br_mis_report: replace type to prefix

OCA-git-bot committed:
Merge PR #3273 into 16.0

Signed-off-by antoniospneto

Commit:
[UPD] Update l10n_br_account_nfe.pot

Commit:
[UPD] Update l10n_br_nfe.pot

Commit:
[UPD] Update l10n_br_nfe_spec.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: l10n-brazil-16.0/l10n-brazil-16.0-l10n_br_nfe_spec
Translate-URL: https://translation.odoo-community.org/projects/l10n-brazil-16-0/l10n-brazil-16-0-l10n_br_nfe_spec/

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: l10n-brazil-16.0/l10n-brazil-16.0-l10n_br_nfe
Translate-URL: https://translation.odoo-community.org/projects/l10n-brazil-16-0/l10n-brazil-16-0-l10n_br_nfe/

antoniospneto committed:
l10n_br_cnpj_search: merge webservice classes

resolve pre-commit:
l10n_br_cnpj_search/models/serpro_webservice.py:19:
[R8180(consider-merging-classes-inherited), SerproWebservice]
Consider merging classes inherited to "l10n_br_cnpj_search.webservice.abstract" from l10n_br_cnpj_search/models/receitaws_webservice.py:11:4.

antoniospneto committed:
l10n_br_fiscal: merge tax_definition classes

antoniospneto committed:
l10n_br_ie_search: merge webservice classes

antoniospneto committed:
l10n_br_fiscal: merge ncm classes

mileo committed:
[FIX][l10n_br_nfe] exported xml not have all fields computed

Signed-off-by: Luis Felipe Miléo <mileo@kmee.com.br>

mileo committed:
[FIX][l10n_br_nfe] certificate company env -> doc

Signed-off-by: Luis Felipe Miléo <mileo@kmee.com.br>

DiegoParadeda committed:
[IMP] l10n_br_fiscal: tax.definition states usability

OCA-git-bot committed:
Merge PR #3329 into 16.0

Signed-off-by antoniospneto

Commit:
[UPD] Update l10n_br_fiscal.pot

Commit:
[UPD] Update l10n_br_account.pot

Commit:
[UPD] Update l10n_br_fiscal_edi.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: l10n-brazil-16.0/l10n-brazil-16.0-l10n_br_account
Translate-URL: https://translation.odoo-community.org/projects/l10n-brazil-16-0/l10n-brazil-16-0-l10n_br_account/

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: l10n-brazil-16.0/l10n-brazil-16.0-l10n_br_fiscal
Translate-URL: https://translation.odoo-community.org/projects/l10n-brazil-16-0/l10n-brazil-16-0-l10n_br_fiscal/

OCA-git-bot committed:
Merge PR #3350 into 16.0

Signed-off-by rvalyi

Commit:
[UPD] Update l10n_br_fiscal.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: l10n-brazil-16.0/l10n-brazil-16.0-l10n_br_fiscal
Translate-URL: https://translation.odoo-community.org/projects/l10n-brazil-16-0/l10n-brazil-16-0-l10n_br_fiscal/

OCA-git-bot committed:
Merge PR #3356 into 16.0

Signed-off-by antoniospneto

Commit:
[UPD] Update l10n_br_account.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: l10n-brazil-16.0/l10n-brazil-16.0-l10n_br_account
Translate-URL: https://translation.odoo-community.org/projects/l10n-brazil-16-0/l10n-brazil-16-0-l10n_br_account/

SottomaiorMacedoTec committed:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 100.0% (65 of 65 strings)

Translation: l10n-brazil-16.0/l10n-brazil-16.0-l10n_br_fiscal_dfe
Translate-URL: https://translation.odoo-community.org/projects/l10n-brazil-16-0/l10n-brazil-16-0-l10n_br_fiscal_dfe/pt_BR/

SottomaiorMacedoTec committed:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 95.4% (21 of 22 strings)

Translation: l10n-brazil-16.0/l10n-brazil-16.0-l10n_br_account_payment_brcobranca
Translate-URL: https://translation.odoo-community.org/projects/l10n-brazil-16-0/l10n-brazil-16-0-l10n_br_account_payment_brcobranca/pt_BR/

SottomaiorMacedoTec committed:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 100.0% (38 of 38 strings)

Translation: l10n-brazil-16.0/l10n-brazil-16.0-l10n_br_account_nfe
Translate-URL: https://translation.odoo-community.org/projects/l10n-brazil-16-0/l10n-brazil-16-0-l10n_br_account_nfe/pt_BR/

SottomaiorMacedoTec committed:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 90.3% (1159 of 1283 strings)

Translation: l10n-brazil-16.0/l10n-brazil-16.0-l10n_br_fiscal
Translate-URL: https://translation.odoo-community.org/projects/l10n-brazil-16-0/l10n-brazil-16-0-l10n_br_fiscal/pt_BR/

SottomaiorMacedoTec committed:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 15.2% (119 of 778 strings)

Translation: l10n-brazil-16.0/l10n-brazil-16.0-l10n_br_account
Translate-URL: https://translation.odoo-community.org/projects/l10n-brazil-16-0/l10n-brazil-16-0-l10n_br_account/pt_BR/

SottomaiorMacedoTec committed:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 85.5% (278 of 325 strings)

Translation: l10n-brazil-16.0/l10n-brazil-16.0-l10n_br_sale
Translate-URL: https://translation.odoo-community.org/projects/l10n-brazil-16-0/l10n-brazil-16-0-l10n_br_sale/pt_BR/

SottomaiorMacedoTec committed:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 87.0% (263 of 302 strings)

Translation: l10n-brazil-16.0/l10n-brazil-16.0-l10n_br_purchase
Translate-URL: https://translation.odoo-community.org/projects/l10n-brazil-16-0/l10n-brazil-16-0-l10n_br_purchase/pt_BR/

SottomaiorMacedoTec committed:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 100.0% (303 of 303 strings)

Translation: l10n-brazil-16.0/l10n-brazil-16.0-l10n_br_stock_account
Translate-URL: https://translation.odoo-community.org/projects/l10n-brazil-16-0/l10n-brazil-16-0-l10n_br_stock_account/pt_BR/

DiegoParadeda committed:
[FIX] l10n_br_nfe: export nfe40_iest to XML and PDF

antoniospneto committed:
[MIG] l10n_br_mis_report: enhance README

DiegoParadeda committed:
[IMP] cancel invoice on fiscal document denial

OCA-git-bot committed:
Merge PR #3343 into 16.0

Signed-off-by marcelsavegnago

Commit:
[UPD] Update l10n_br_mis_report.pot

OCA-git-bot committed:
[BOT] post-merge updates

marcelsavegnago committed:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 52.6% (158 of 300 strings)

Translation: l10n-brazil-16.0/l10n-brazil-16.0-l10n_br_cnab_structure
Translate-URL: https://translation.odoo-community.org/projects/l10n-brazil-16-0/l10n-brazil-16-0-l10n_br_cnab_structure/pt_BR/

marcelsavegnago committed:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 90.4% (1160 of 1283 strings)

Translation: l10n-brazil-16.0/l10n-brazil-16.0-l10n_br_fiscal
Translate-URL: https://translation.odoo-community.org/projects/l10n-brazil-16-0/l10n-brazil-16-0-l10n_br_fiscal/pt_BR/

marcelsavegnago committed:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 18.5% (169 of 913 strings)

Translation: l10n-brazil-16.0/l10n-brazil-16.0-l10n_br_nfe
Translate-URL: https://translation.odoo-community.org/projects/l10n-brazil-16-0/l10n-brazil-16-0-l10n_br_nfe/pt_BR/

marcelsavegnago committed:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 100.0% (65 of 65 strings)

Translation: l10n-brazil-16.0/l10n-brazil-16.0-l10n_br_fiscal_dfe
Translate-URL: https://translation.odoo-community.org/projects/l10n-brazil-16-0/l10n-brazil-16-0-l10n_br_fiscal_dfe/pt_BR/

marcelsavegnago committed:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 100.0% (75 of 75 strings)

Translation: l10n-brazil-16.0/l10n-brazil-16.0-l10n_br_fiscal_certificate
Translate-URL: https://translation.odoo-community.org/projects/l10n-brazil-16-0/l10n-brazil-16-0-l10n_br_fiscal_certificate/pt_BR/

marcelsavegnago committed:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 100.0% (96 of 96 strings)

Translation: l10n-brazil-16.0/l10n-brazil-16.0-l10n_br_fiscal_closing
Translate-URL: https://translation.odoo-community.org/projects/l10n-brazil-16-0/l10n-brazil-16-0-l10n_br_fiscal_closing/pt_BR/

marcelsavegnago committed:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 25.1% (124 of 493 strings)

Translation: l10n-brazil-16.0/l10n-brazil-16.0-l10n_br_account_payment_order
Translate-URL: https://translation.odoo-community.org/projects/l10n-brazil-16-0/l10n-brazil-16-0-l10n_br_account_payment_order/pt_BR/

marcelsavegnago committed:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 100.0% (40 of 40 strings)

Translation: l10n-brazil-16.0/l10n-brazil-16.0-l10n_br_zip
Translate-URL: https://translation.odoo-community.org/projects/l10n-brazil-16-0/l10n-brazil-16-0-l10n_br_zip/pt_BR/

marcelsavegnago committed:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 100.0% (364 of 364 strings)

Translation: l10n-brazil-16.0/l10n-brazil-16.0-l10n_br_hr
Translate-URL: https://translation.odoo-community.org/projects/l10n-brazil-16-0/l10n-brazil-16-0-l10n_br_hr/pt_BR/

marcelsavegnago committed:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 86.7% (282 of 325 strings)

Translation: l10n-brazil-16.0/l10n-brazil-16.0-l10n_br_sale
Translate-URL: https://translation.odoo-community.org/projects/l10n-brazil-16-0/l10n-brazil-16-0-l10n_br_sale/pt_BR/

marcelsavegnago committed:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 100.0% (303 of 303 strings)

Translation: l10n-brazil-16.0/l10n-brazil-16.0-l10n_br_stock_account
Translate-URL: https://translation.odoo-community.org/projects/l10n-brazil-16-0/l10n-brazil-16-0-l10n_br_stock_account/pt_BR/

marcelsavegnago committed:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 87.7% (265 of 302 strings)

Translation: l10n-brazil-16.0/l10n-brazil-16.0-l10n_br_purchase
Translate-URL: https://translation.odoo-community.org/projects/l10n-brazil-16-0/l10n-brazil-16-0-l10n_br_purchase/pt_BR/

marcelsavegnago committed:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 0.7% (8 of 1135 strings)

Translation: l10n-brazil-16.0/l10n-brazil-16.0-l10n_br_nfe_spec
Translate-URL: https://translation.odoo-community.org/projects/l10n-brazil-16-0/l10n-brazil-16-0-l10n_br_nfe_spec/pt_BR/

marcelsavegnago committed:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 22.3% (174 of 778 strings)

Translation: l10n-brazil-16.0/l10n-brazil-16.0-l10n_br_account
Translate-URL: https://translation.odoo-community.org/projects/l10n-brazil-16-0/l10n-brazil-16-0-l10n_br_account/pt_BR/

antoniospneto committed:
[REF] l10n_br_account: remove dead code

OCA-git-bot committed:
Merge PR #3372 into 16.0

Signed-off-by antoniospneto

OCA-git-bot committed:
[BOT] post-merge updates

corredato committed:
[IMP] l10n_br_fiscal: icms fcp and difal destination comments

OCA-git-bot committed:
Merge PR #3308 into 16.0

Signed-off-by antoniospneto

OCA-git-bot committed:
[BOT] post-merge updates

SottomaiorMacedoTec committed:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 100.0% (22 of 22 strings)

Translation: l10n-brazil-16.0/l10n-brazil-16.0-l10n_br_account_payment_brcobranca
Translate-URL: https://translation.odoo-community.org/projects/l10n-brazil-16-0/l10n-brazil-16-0-l10n_br_account_payment_brcobranca/pt_BR/

WesleyOliveira98 committed:
[FIX] l10n_br_account: fix _compute_fiscal_document_ids

DiegoParadeda committed:
[IMP] l10n_br_account: document deny tests

OCA-git-bot committed:
Merge PR #3378 into 16.0

Signed-off-by marcelsavegnago

OCA-git-bot committed:
[BOT] post-merge updates

DiegoParadeda committed:
[FIX] l10n_br_fiscal_edi: rename refactored reports

rvalyi committed:
[FIX] l10n_br_fiscal: fix field defaults/labels

[FIX] l10n_br_fiscal: default currency_id fix

rvalyi committed:
[FIX] l10n_br_fiscal_edi: fix warnings

rvalyi commented:
faltou adaptar o script de migração, vou botar em rascunho até la.

OCA-git-bot committed:
Merge PR #3400 into 16.0

Signed-off-by yvaucher

Commit:
[UPD] Update l10n_br_fiscal_edi.pot

OCA-git-bot committed:
[BOT] post-merge updates

OCA-git-bot committed:
Merge PR #3399 into 16.0

Signed-off-by rvalyi

OCA-git-bot committed:
[BOT] post-merge updates

antoniospneto commented:
@rvalyi aproveitando, eu to refatorando o script de migração do cnab na 14.0, se puder esperar um pouco, acho que vele a pena ver a minha proposta

OCA-git-bot committed:
Merge PR #3393 into 16.0

Signed-off-by rvalyi

OCA-git-bot committed:
[BOT] post-merge updates

OCA-git-bot committed:
Merge PR #3395 into 16.0

Signed-off-by rvalyi

OCA-git-bot committed:
[BOT] post-merge updates

OCA-git-bot committed:
Merge PR #3405 into 16.0

Signed-off-by rvalyi

Commit:
[UPD] Update l10n_br_fiscal.pot

Commit:
[UPD] Update l10n_br_account.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: l10n-brazil-16.0/l10n-brazil-16.0-l10n_br_account
Translate-URL: https://translation.odoo-community.org/projects/l10n-brazil-16-0/l10n-brazil-16-0-l10n_br_account/

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: l10n-brazil-16.0/l10n-brazil-16.0-l10n_br_fiscal
Translate-URL: https://translation.odoo-community.org/projects/l10n-brazil-16-0/l10n-brazil-16-0-l10n_br_fiscal/

kaynnan committed:
[IMP] l10n_br_fiscal_closing: add document_cte_ids

rvalyi committed:
[FIX] l10n_br_account: avoid duplicated labels

rvalyi committed:
[FIX] l10n_br_purchase: fix field name

rvalyi committed:
[REM] l10n_br_fiscal_closing: rm useless onchange

rvalyi committed:
[MIG] l10n_br_nfe: deprecated flush and cache meth

rvalyi committed:
[REF] spec_driven_model: rm dead code; + XML tests

Tiago370 committed:
[IMP] l10n_br_cnpj_search: change vcr mocks for unittest mocks.

OCA-git-bot committed:
Merge PR #3385 into 16.0

Signed-off-by rvalyi

Commit:
[UPD] Update l10n_br_fiscal.pot

Commit:
[UPD] Update l10n_br_account.pot

Commit:
[UPD] Update l10n_br_purchase.pot

Commit:
[UPD] Update l10n_br_sale.pot

Commit:
[UPD] Update l10n_br_stock_account.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: l10n-brazil-16.0/l10n-brazil-16.0-l10n_br_account
Translate-URL: https://translation.odoo-community.org/projects/l10n-brazil-16-0/l10n-brazil-16-0-l10n_br_account/

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: l10n-brazil-16.0/l10n-brazil-16.0-l10n_br_fiscal
Translate-URL: https://translation.odoo-community.org/projects/l10n-brazil-16-0/l10n-brazil-16-0-l10n_br_fiscal/

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: l10n-brazil-16.0/l10n-brazil-16.0-l10n_br_stock_account
Translate-URL: https://translation.odoo-community.org/projects/l10n-brazil-16-0/l10n-brazil-16-0-l10n_br_stock_account/

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: l10n-brazil-16.0/l10n-brazil-16.0-l10n_br_sale
Translate-URL: https://translation.odoo-community.org/projects/l10n-brazil-16-0/l10n-brazil-16-0-l10n_br_sale/

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: l10n-brazil-16.0/l10n-brazil-16.0-l10n_br_purchase
Translate-URL: https://translation.odoo-community.org/projects/l10n-brazil-16-0/l10n-brazil-16-0-l10n_br_purchase/

OCA-git-bot committed:
Merge PR #3428 into 16.0

Signed-off-by rvalyi

OCA-git-bot committed:
[BOT] post-merge updates

OCA-git-bot committed:
Merge PR #3436 into 16.0

Signed-off-by rvalyi

Commit:
[UPD] Update l10n_br_fiscal.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: l10n-brazil-16.0/l10n-brazil-16.0-l10n_br_fiscal
Translate-URL: https://translation.odoo-community.org/projects/l10n-brazil-16-0/l10n-brazil-16-0-l10n_br_fiscal/

Tiago370 committed:
[IMP] l10n_br_cnpj_search: solves cnpj duplication problem in tests

Tiago370 committed:
[IMP] l10n_br_cnpj_search: transfer the ws mock to comon

Tiago370 committed:
[IMP] l10n_br_cnpj_search: transfer the serpro mock to common

OCA-git-bot committed:
Merge PR #3438 into 16.0

Signed-off-by mileo

OCA-git-bot committed:
[BOT] post-merge updates

CristianoMafraJunior committed:
[MIG] l10n_br_account_nfe: _render_qweb_* migration

CristianoMafraJunior committed:
[FIX] l10n_br_nfe: name template

OCA-git-bot committed:
Merge PR #3437 into 16.0

Signed-off-by rvalyi

OCA-git-bot committed:
[BOT] post-merge updates

OCA-git-bot committed:
Merge PR #3439 into 16.0

Signed-off-by antoniospneto

OCA-git-bot committed:
[BOT] post-merge updates

Tiago370 committed:
[IMP] l10n_br_ie_search: change vcr mocks for unittest mocks.

OCA-git-bot committed:
Merge PR #3440 into 16.0

Signed-off-by antoniospneto

Commit:
[UPD] Update l10n_br_account.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: l10n-brazil-16.0/l10n-brazil-16.0-l10n_br_account
Translate-URL: https://translation.odoo-community.org/projects/l10n-brazil-16-0/l10n-brazil-16-0-l10n_br_account/

OCA-git-bot committed:
Merge PR #3448 into 16.0

Signed-off-by mileo

OCA-git-bot committed:
[BOT] post-merge updates

Tiago370 committed:
[FIX] equal Estadual Inscription detection

Tiago370 committed:
[IMP] add tests.

OCA-git-bot committed:
Merge PR #3450 into 16.0

Signed-off-by mileo

OCA-git-bot committed:
[BOT] post-merge updates

rvalyi committed:
[REF] spec_driven_model: pre-commit fixes

OCA-git-bot committed:
Merge PR #3444 into 16.0

Signed-off-by rvalyi

Commit:
[UPD] Update l10n_br_account_nfe.pot

Commit:
[UPD] Update l10n_br_nfe_spec.pot

Commit:
[UPD] Update l10n_br_nfe.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: l10n-brazil-16.0/l10n-brazil-16.0-l10n_br_nfe_spec
Translate-URL: https://translation.odoo-community.org/projects/l10n-brazil-16-0/l10n-brazil-16-0-l10n_br_nfe_spec/

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: l10n-brazil-16.0/l10n-brazil-16.0-l10n_br_nfe
Translate-URL: https://translation.odoo-community.org/projects/l10n-brazil-16-0/l10n-brazil-16-0-l10n_br_nfe/

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: l10n-brazil-16.0/l10n-brazil-16.0-l10n_br_account_nfe
Translate-URL: https://translation.odoo-community.org/projects/l10n-brazil-16-0/l10n-brazil-16-0-l10n_br_account_nfe/

OCA-git-bot committed:
Merge PR #3459 into 16.0

Signed-off-by rvalyi

Commit:
[UPD] Update l10n_br_nfe.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: l10n-brazil-16.0/l10n-brazil-16.0-l10n_br_nfe
Translate-URL: https://translation.odoo-community.org/projects/l10n-brazil-16-0/l10n-brazil-16-0-l10n_br_nfe/

OCA-git-bot committed:
Merge PR #3457 into 16.0

Signed-off-by rvalyi

Commit:
[UPD] Update l10n_br_fiscal_closing.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: l10n-brazil-16.0/l10n-brazil-16.0-l10n_br_fiscal_closing
Translate-URL: https://translation.odoo-community.org/projects/l10n-brazil-16-0/l10n-brazil-16-0-l10n_br_fiscal_closing/

Tiago370 committed:
[IMP] l10n_br_cnpj_search: default_get from res.partner

OCA-git-bot committed:
Merge PR #3454 into 16.0

Signed-off-by rvalyi

Commit:
[UPD] Update spec_driven_model.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: l10n-brazil-16.0/l10n-brazil-16.0-spec_driven_model
Translate-URL: https://translation.odoo-community.org/projects/l10n-brazil-16-0/l10n-brazil-16-0-spec_driven_model/

OCA-git-bot committed:
Merge PR #3289 into 16.0

Signed-off-by rvalyi

Commit:
[UPD] Update l10n_br_crm_cnpj_search.pot

OCA-git-bot committed:
[BOT] post-merge updates

CristianoMafraJunior committed:
[FIX] l10n_br_fiscal: tree view field

OCA-git-bot committed:
Merge PR #3468 into 16.0

Signed-off-by mileo

OCA-git-bot committed:
[BOT] post-merge updates

CristianoMafraJunior committed:
[FIX] l10n_br_account: tree view field

CristianoMafraJunior committed:
[FIX] l10n_br_fiscal_edi: tree view field

CristianoMafraJunior committed:
[FIX] l10n_br_fiscal: tree view field

OCA-git-bot committed:
Merge PR #3470 into 16.0

Signed-off-by antoniospneto

OCA-git-bot committed:
[BOT] post-merge updates

crsilveira committed:
[FIX] l10n_br_fiscal: ncm starting with 09 conflicts with service type

OCA-git-bot committed:
Merge PR #3473 into 16.0

Signed-off-by antoniospneto

OCA-git-bot committed:
[BOT] post-merge updates

CristianoMafraJunior committed:
[FIX] l10n_br_base: form padding 0

OCA-git-bot committed:
Merge PR #3477 into 16.0

Signed-off-by rvalyi

OCA-git-bot committed:
[BOT] post-merge updates

antoniospneto committed:
[REF] l10n_br_account_payment_order: Mig Script

mbcosta committed:
[FIX+MIG] l10n_br_account_payment_order:Mig Script

mbcosta committed:
[REM] l10n_br_account_payment_order:Fields / Views

mbcosta committed:
[UPD]l10n_br_account_payment_brcobranca:Pre-Commit

mbcosta committed:
[UPD] l10n_br_account_payment_order: Pre-Commit

mbcosta committed:
[FIX+MIG] l10n_br_account_payment_order: Views v16

mbcosta commented:
@rvalyi @antoniospneto atualizei o script de migração e fiz outras pequenas alterações necessárias, segue o que foi feito:

- Copiei diretamente o arquivo [l10n_br_account_payment_order/migrations/14.0.6.0.0/post-migration.py da v14](https://github.com/OCA/l10n-brazil/blob/14.0/l10n_br_account_payment_order/migrations/14.0.6.0.0/post-migration.py ) colocando o Antonio como autor, mas foi preciso corrigir o script porque os **Códigos de Retorno que geram Baixa/Liquidação** e o **Código de Carteira** não estavam sendo copiados, pelo menos na v16 ( na v14 é preciso confirmar)

![image](https://github.com/user-attachments/assets/ef088040-77f1-443a-9324-246dad28bdfc)

![image](https://github.com/user-attachments/assets/766ae440-d0c6-4ccc-8c18-2389103ce77e)

![image](https://github.com/user-attachments/assets/631e317b-f1d7-49fe-9668-cd476ef45cfc)

![image](https://github.com/user-attachments/assets/32f4baac-8ba9-4410-9ee5-1cf3151ea65b)

Um ponto aqui, até onde vi é desnecessário tanto na v14 quanto na v16 criar os campos via script como foi feito no [l10n_br_account_payment_order/migrations/14.0.6.0.0/pre-migration.py](https://github.com/OCA/l10n-brazil/blob/14.0/l10n_br_account_payment_order/migrations/14.0.6.0.0/pre-migration.py) o **Openupgradelib** tem uma função para adicionar campos [def add_fields](https://github.com/OCA/openupgradelib/blob/master/openupgradelib/openupgrade.py#L2910) mas no docstring está dizendo que a partir da v12 isso seria desnecessário e não tive erros ao não usar

![image](https://github.com/user-attachments/assets/eef61700-0766-4d20-9cd6-e2afc2c1c659)

E para não rodar o script em Banco de Dados já migrados é verificado se exite algum **Modo de Pagamento** com o campo **sending_code_id** preenchido, o campo é fundamental para o uso do CNAB, e para testar é possível   simular mudando a versão do modulo para **16.0.3.0.0** e copiando o mesmo script da versão **16.0.2.0.0**.

- Com esse script de migração foi possível remover campos e visões que se tornaram obsoletos, fazendo boa parte do que foi feito no PR https://github.com/OCA/l10n-brazil/pull/3365 porém não foi possível remover os objetos que foram unificados isso gera erro porque o programa parece buscar os antigos XML que tem esses objetos, depois de rodar o script e depois da 16.0.2.0.0 isso deve deixar de causar erros mas para evitar o problema que foi visto na v14, migrar múltiplas versões de uma vez https://github.com/OCA/l10n-brazil/issues/3408 eu recomendo ou deixar para remover os objetos daqui 6 meses ou apenas na v17, deixei um TODO com o LOG de erro completo para deixar claro o problema

```bash
# TODO: Ao remover os objetos que ficaram obsoletos com a unificação dos Códigos
#  no l10n_br_cnab.code acontece o erro abaixo, isso deixa de acontecer em
#  versões posteriores a 16.0.2.0.0, porém para evitar problemas como o que ocorreu
#  na v14 será melhor aguardar mesmo que por 6 meses ou mesmo apenas na migração
#  para a v17 para remover esses objetos e assim evitar qualquer possibilidade
#  de problema com migrações
# ERROR db odoo.modules.registry: Failed to load registry
# Traceback (most recent call last):
#  File "/usr/local/lib/python3.10/site-packages/odoo/tools/convert.py",
#  line 698, in _tag_root
#    f(rec)
#  File "/usr/local/lib/python3.10/site-packages/odoo/tools/convert.py", line 515,
#  in _tag_record
#    record = env['ir.model.data']._load_xmlid(xid)
#  File "/usr/local/lib/python3.10/site-packages/odoo/addons/base/models/ir_model.py",
#  line 2162, in _load_xmlid
#    record = self.env.ref(xml_id, raise_if_not_found=False)
#  File "/usr/local/lib/python3.10/site-packages/odoo/api.py", line 600, in ref
#    record = self[res_model].browse(res_id)
#  File "/usr/local/lib/python3.10/site-packages/odoo/api.py", line 550, in __getitem__
#    return self.registry[model_name](self, (), ())
#  File "/usr/local/lib/python3.10/site-packages/odoo/modules/registry.py",
#  line 190, in __getitem__
#    return self.models[model_name]
# KeyError: 'l10n_br_cnab.mov.instruction.code'
```

- Foi preciso atualizar o PR devido as atualizações do pre-commit, foi feito em commits em separado

- Parece que na migração https://github.com/OCA/l10n-brazil/pull/3222 faltou adaptar as Visões para as alterações que foram feitas na v16, algumas referencias sobre a mudança

```bash
https://github.com/odoo/odoo/issues/109447

https://github.com/odoo/odoo/pull/102119

https://github.com/odoo/odoo/pull/102190/files

Os PRs estão dentro desse "bash" para evitar uma ligação direta com esses PRs e acabarem "poluindo" os PRs, segue images
```
![image](https://github.com/user-attachments/assets/0b3d6454-cea7-4127-84f2-8420a9a69ba2)

![image](https://github.com/user-attachments/assets/605c96e8-e4ce-4edb-8e14-92f8f92a4d24)

![image](https://github.com/user-attachments/assets/111c2a82-6201-43f5-9610-72d27ebd579c)

O problema que acontecia aqui 

**Modo de Pagamento**

Antes do PR 
![image](https://github.com/user-attachments/assets/73838124-09af-4a2f-a765-5bc2e9df119a)

![image](https://github.com/user-attachments/assets/e68a9e52-47c0-4aca-bf6a-2cbe0eff8f70)
 
Com esse PR
![image](https://github.com/user-attachments/assets/4d724e3f-270b-4ded-bf83-09016fb64910)


**Linhas de Movimentações Financeiras**

Antes do PR 
![image](https://github.com/user-attachments/assets/19bee9be-b7ea-4069-8c2a-9e3adb40eeb8)

Com esse PR
![image](https://github.com/user-attachments/assets/5f7f3f6e-c19a-4832-9443-d9de0301c870)

Se acreditarem que vai levar um tempo para Aprovar esse PR posso ver de extrair esse commit que corrige essas Visões, aqui é necessário para permitir a Revisão.

Com o que foi feito no PR acredito e peço para alterar o PR de **Rascunho/Draft** para **Pronto para Revisão/Ready to Review**

SottomaiorMacedoTec committed:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 100.0% (303 of 303 strings)

Translation: l10n-brazil-16.0/l10n-brazil-16.0-l10n_br_stock_account
Translate-URL: https://translation.odoo-community.org/projects/l10n-brazil-16-0/l10n-brazil-16-0-l10n_br_stock_account/pt_BR/

SottomaiorMacedoTec committed:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 100.0% (5 of 5 strings)

Translation: l10n-brazil-16.0/l10n-brazil-16.0-spec_driven_model
Translate-URL: https://translation.odoo-community.org/projects/l10n-brazil-16-0/l10n-brazil-16-0-spec_driven_model/pt_BR/

SottomaiorMacedoTec committed:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 100.0% (98 of 98 strings)

Translation: l10n-brazil-16.0/l10n-brazil-16.0-l10n_br_fiscal_closing
Translate-URL: https://translation.odoo-community.org/projects/l10n-brazil-16-0/l10n-brazil-16-0-l10n_br_fiscal_closing/pt_BR/

SottomaiorMacedoTec committed:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 100.0% (36 of 36 strings)

Translation: l10n-brazil-16.0/l10n-brazil-16.0-l10n_br_account_nfe
Translate-URL: https://translation.odoo-community.org/projects/l10n-brazil-16-0/l10n-brazil-16-0-l10n_br_account_nfe/pt_BR/

DiegoParadeda committed:
[FIX] l10n_br_account_payment_order: cnab config copy

OCA-git-bot committed:
Merge PR #3489 into 16.0

Signed-off-by antoniospneto

OCA-git-bot committed:
[BOT] post-merge updates

rvalyi commented:
subsituido por #3511
